nslsii/areadetector/xspress3.py

```diff
@@ -0,0 +1,838 @@
+        if mcaroi_prefix_match is None:
+            raise ValueError(
+                f"mcaroi prefix '{prefix}' does not match the expected pattern `{self.mcaroi_prefix_re.pattern}`"
+            )
+        self.mcaroi_number = int(mcaroi_prefix_match.group("mcaroi_number"))
+
+    def configure_mcaroi(self, *, min_x, size_x, roi_name=None, use=True):
+        """
+        Configure the details of an MCAROI.
+
+        Parameters
+        ----------
+        min_x: int
+            the starting bin? for the roi
+        size_x: int
+            the width in bins? for the roi
+        roi_name: str, optional
+        use: bool, defaults to True
+        """
+
+        logger.debug(
+            "configuring Xspress3 MCAROI '%s': name '%s' min_x '%d' size_x '%d' use '%s'",
+            self.prefix,
+            roi_name,
+            min_x,
+            size_x,
+            use,
+        )
+        self.min_x.put(int(min_x))
+        self.size_x.put(int(size_x))
+        if roi_name:
+            self.roi_name.put(roi_name)
+        self.use.put(use)
+
+    # remove in favor of configure_mcaroi
+    def configure_roi(self, ev_min, ev_size):
+        """Configure the MCAROI with min and size eV
+
+        Parameters
+        ----------
+        ev_min : int
+            minimum electron volts for ROI
+        ev_size : int
+            ROI size (width) in electron volts
+        """
+        ev_min = int(ev_min)
+        ev_size = int(ev_size)
+
+        # assume if this ROI is being configured
+        # that it should be read, meaning the
+        # "use" PV must be set to 1
+        use_roi = 1
+        configuration_changed = any(
+            [
+                self.min_x.get() != ev_min,
+                self.size_x.get() != ev_size,
+                self.use.get() != use_roi,
+            ]
+        )
+
+        if configuration_changed:
+            logger.debug(
+                "Setting up Xspress3 ROI: name=%s ev_min=%s ev_size=%s "
+                "use=%s prefix=%s channel=%s",
+                self.name,
+                ev_min,
+                ev_size,
+                use_roi,
+                self.prefix,
+                # self.parent is the ?? class
+                # self.parent.parent is the ?? class
+                # TODO: I don't like the assumption that self has a parent
+                self.parent.parent.channel_num,
+            )
+
+            self.min_x.put(ev_min)
+            self.size_x.put(ev_size)
+            self.use.put(use_roi)
+        else:
+            # nothing has changed
+            pass
+
+    def clear(self):
+        """Clear and disable this ROI"""
+        # it is enough to just disable the ROI
+        # self.min_x.put(0)
+        # self.size_x.put(0)
+        self.use.put(0)
+
+
+class Sca(ADBase):
+    # includes Dead Time correction, for example
+    # sca numbers go from 0 to 10
+    clock_ticks = Cpt(EpicsSignalRO, "0:Value_RBV")
```

Comment:
Is it worth putting kinds on these?

Reply:
What kinds would you like?

Reply:
https://github.com/epics-modules/xspress3/blob/7ffb19ad224621fed179b8fb82723eb0590d0b85/INSTALL_Standalone.rst#deadtime-values-c[…]ns-and-eventwidth for some documentation on what these are.